benches/bench.rs

```diff
@@ -77,31 +80,85 @@ fn as_hash(inp: u32) -> [u8; 32] {
     out
 }
 
+fn as_hash160(inp: u32) -> [u8; 20] {
+    let mut out = [0; 20];
+    out[0..4].copy_from_slice(&inp.to_le_bytes());
+    out
+}
+
 fn setup_chain_state<'a>(scaling: u32, headers_db: &'a dyn HeadersDB) -> MarfedKV {
     let pre_initialized_path = format!("/tmp/clarity_bench_{}.marf", scaling);
     let out_path = "/tmp/clarity_bench_last.marf";
 
     if fs::metadata(&pre_initialized_path).is_err() {
         let marf = MarfedKV::open(&pre_initialized_path, None).unwrap();
         let mut clarity_instance = ClarityInstance::new(false, marf, ExecutionCost::max_value());
+
+        let p1 = QualifiedContractIdentifier::parse("S1G2081040G2081040G2081040G208105NK8PE5.c1").unwrap().issuer;
+        let principals: [PrincipalData; 2] = [
+            p1.into(), StandardPrincipalData(0, as_hash160(2)).into(),
+        ];
+
         let mut conn = clarity_instance.begin_test_genesis_block(
             &StacksBlockId::sentinel(),
             &StacksBlockId(as_hash(0)),
             &*headers_db,
             &NULL_BURN_STATE_DB,
         );
 
+        // mint and scale marf in genesis block
         conn.as_transaction(|tx| {
-            for j in 0..scaling {
-                tx.with_clarity_db(|db| {
+            // minting
+            tx.with_clarity_db(|db| {
+                principals.iter().for_each(|p| {
+                    let mut stx_account = db.get_stx_balance_snapshot_genesis(&p);
+                    stx_account.credit(1_000_000);
+                    stx_account.save();
+                });
+                Ok(())
+            }).unwrap();
+
+            // scaling
+            tx.with_clarity_db(|db| {
+                (0..100000).for_each(|j| {
                     db.put(format!("key{}", j).as_str(), &Value::none());
+                });
+                Ok(())
+            })
+            .unwrap();
+        });
+
+        conn.commit_to_block(&StacksBlockId(as_hash(0)));
+
+        // append more blocks
+        let blocks: Vec<_> = (0..=NUM_BLOCKS)
+            .into_iter()
+            .map(|i| StacksBlockId(as_hash(i)))
+            .collect();
+
+        for ix in 1..=(blocks.len() - 1) {
```

Comment:
this loop evaluates 58 times - is that what you want?

Reply:
It actually evaluates 60 times since `blocks.len()` is 61. That being said, I'll make this clearer by changing it to:

```Rust
for ix in 1..blocks.len() { ... }
```

Reply:
ahh I missed the equal sign in both iterators - makes sense to me!